### PR TITLE
ci: generate junit reports for unit tests

### PR DESCRIPTION
**Pull request by yakimant**
We want to make failed tests more visible without reading the logs and also use Jenkins features, like for tracking test history.
We need to have an integration between `go test` and Jenkins through junit xml files. This is an industry standard for tests-ci integration.

This PR adds too instalation and junit reports generation.

### CODE REVIEW

**Comment by yakimant** [comment]
@richard-ramos, @andremedeiros, there is an issue of importing `go test` results into Jenkins.

The problem is: `An invalid XML character (Unicode: 0x1b) was found in the CDATA section.`

You can see it in this example:
```
=== RUN   TestBackendStartNodeConcurrently
[32mINFO [0m[08-09|15:40:42.990] Status backend initialized               [32mbackend[0m=geth [32mversion[0m= [32mcommit[0m= [32mIpfsGatewayURL[0m=
[32mINFO [0m[08-09|15:40:42.992] status-go version details                [32mpackage[0m=status-go/api.GethStatusBackend [32mversion[0m= [32mcommit[0m=
[32mINFO [0m[08-09|15:40:43.022] Starting peer-to-peer node               [32minstance[0m=test/darwin-arm64/go1.20.6
[32mINFO [0m[08-09|15:40:43.046] Successful start 
[32mINFO [0m[08-09|15:40:43.046] Handlers registered                      [32mpackage[0m=status-go/api.GethStatusBackend
[32mINFO [0m[08-09|15:40:43.046] status-go version details                [32mpackage[0m=status-go/api.GethStatusBackend [32mversion[0m= [32mcommit[0m=
[32mINFO [0m[08-09|15:40:43.051] New local node record                    [32mseq[0m=1,691,588,232,805 [32mid[0m=c865c27c3a7402df [32mip[0m=127.0.0.1 [32mudp[0m=0 [32mtcp[0m=55139
[32mINFO [0m[08-09|15:40:43.051] Started P2P networking                   [32mself[0m="enode://6acc2610b07e649e04d38864ad5ffa57f08599c92e66afda69b5d7b18b56c91d27478531f4fd78067b09501a5505df6f03daea5d38a59b477b732e38c141c8fb@127.0.0.1:55139?discport=0"
--- PASS: TestBackendStartNodeConcurrently (2.25s)
```
Is it this on purpose? Can it be cleaned out?

As an alternative - I can clean the output.

**Comment by yakimant** [comment]
Ok, it's an excape character for terminal colours: `<0x1b>[32m` and `<0x1b>[0m`.
Can we disable terminal colours easily with some `go test` argument?

**Comment by richard-ramos** [comment]
Quickly looking at the code, to remove the format, we should call https://github.com/status-im/status-go/blob/develop/logutils/override.go#L46 passing a `false` to the `terminal` parameter, so that means modifying `utils.Init()` so it can receive an optional value to indicate whether you want colors in the terminal or not (maybe depending on an environment variable), and modify the tests so this function is called (only few tests call that function)

Is the content of the tests required? i wonder if you can conditionally remove the `-v` flag of go tests, so these logs are not shown, and instead only display the test results.

**Comment by yakimant** [comment]
Thanks @richard-ramos!

The question about `-v` is to you and other developers. Do you need it for tests debug for example?
Both on CI and local.
Do you want colors and verbosity localy, but fine without verbosity on CI?

I applied escape and terminal colors cleanup instead of modifying the code, works well:
- Jenkins Job: https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3871/7/testReport/github/
- Also in Blue Ocean: https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-3871/7/tests
- Test results trend: https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3871/

BTW, few protocol tests are using a lot of time, probably worth to optimize:
https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3871/7/testReport/github/com_status-im_status-go_protocol/

**Comment by richard-ramos** [comment]
> The question about -v is to you and other developers. Do you need it for tests debug for example?
Both on CI and local.
Do you want colors and verbosity localy, but fine without verbosity on CI?

cc: @cammellos  @alaibe @jrainville @Samyoul

**Comment by cammellos** [comment]
> > The question about -v is to you and other developers. Do you need it for tests debug for example?
> > Both on CI and local.
> > Do you want colors and verbosity localy, but fine without verbosity on CI?
> 
> cc: @cammellos @alaibe @jrainville @Samyoul

I am fine with either, I search through the file/logs with an editor, so I wouldn't personally make use of them, but nothing against it.

**Comment by yakimant** [comment]
Removing `-v` is not an option, there is no output at all.
To have a proper solution in go to enable/disable I'm checking the code now.

File: _assets/ci/Jenkinsfile.tests [hunk 1]
```diff
@@ -70,6 +70,7 @@ pipeline {
 
     stage('Unit Tests') {
       steps { script {
+        nix.shell('make install-junit-report', pure: false)
```
Comment (jakubgs) [inline _assets/ci/Jenkinsfile.tests]: There is no need for this. Add it to `shell.nix`.

File: logutils/override.go [hunk 2]
```diff
@@ -48,6 +48,9 @@ func OverrideRootLog(enabled bool, levelStr string, fileOpts FileOptions, termin
 		disableRootLog()
 		return nil
 	}
+	if os.Getenv("CI") == "true" {
```
Comment (jakubgs) [inline logutils/override.go:51]: Nice.

**Review by jakubgs — APPROVED** [review]
Looks good to me. Other than installing manually the dependency in `Jenkinsfile`.

This dependency is already available in Nix, add it there.

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**